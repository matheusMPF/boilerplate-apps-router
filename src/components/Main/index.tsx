import * as S from './styles'

const Main = () => {
  return (
    <S.wrapper>
      <h1>Hello world</h1>
    </S.wrapper>
  )
}

export default Main
