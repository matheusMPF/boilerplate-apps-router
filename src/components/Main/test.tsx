import { render, screen } from '@testing-library/react'
import Main from './index'

describe('<Main />', () => {
  it('should render the heading', () => {
    // Passo 1. Renderizar o componente
    const { container } = render(<Main />)
    // Passo 2. Buscar o elemento e verificar a existÃªncia dele
    expect(
      screen.getByRole('heading', { name: /Hello world/i })
    ).toBeInTheDocument()

    //Gerar snapshot -> gera uma pasta snapshot na raiz do teste. Apenas se quisermos uma estrutura muito especifica
    expect(container.firstChild).toMatchSnapshot()
  })
})
