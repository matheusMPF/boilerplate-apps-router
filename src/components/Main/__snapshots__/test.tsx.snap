// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
<main
  class="sc-blHHSb bZInSn"
>
  <h1>
    Hello world
  </h1>
</main>
`;
